src/reducers/editorialWorkflow.js

```diff
@@ -39,7 +39,8 @@ const unpublishedEntries = (state = null, action) => {
       return state.setIn(['pages', 'isFetching'], true);
 
     case UNPUBLISHED_ENTRIES_SUCCESS:
-      return state.withMutations((map) => {
+      // reset the state when unpublished entries are fetched anew.
+      return fromJS({ entities: {}, pages: {} }).withMutations((map) => {
```

Comment:
Is this line fixing the reappearing unpublished deleted entries on your end? I'm still seeing the issue even with this change.